ring/ring.go

```diff
@@ -210,22 +210,28 @@ func (r *Ring) Describe(ch chan<- *prometheus.Desc) {
 	ch <- r.numTokensDesc
 }
 
-// Collect implements prometheus.Collector.
-func (r *Ring) Collect(ch chan<- prometheus.Metric) {
-	r.mtx.RLock()
-	defer r.mtx.RUnlock()
-
+func countTokens(tokens []TokenDesc) (map[string]uint32, map[string]uint32) {
```

Comment:
Return type of this is non-obvious (even after looking at the implementation & variable names—what's `numTokens`? what's `owned`?). Worth adding a comment saying what it's returning.